### PR TITLE
v1.4: Removal of persons and relevant refactorings

### LINKED ISSUE #129: Remove all Persons
Remove everything related to `Person` -- commands, parsers, models, test cases, everything. 
Self note to becareful when removing test cases as require updating system tests and UI tests in which case, replace all Persons to Accounts.

Once this is done, remove everything related in UG/DG as well.

### LINKED ISSUE #219: login command doesnt work (severity.Low)
screenshot-1541147125_732x139

Unable to login using the login function `li`. 

Seems to be a clash with the command used to list the entries will also uses the command work `li` (2.5.4).

**severity.Low** because main login command still works

<hr>

**Reported by:** @ongspxm
**Severity:** `Low`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#61]</sub>

### LINKED ISSUE #225: [PE Dry Run]deregistering currently logged in account
screenshot-1541147674_1578x214

ran command `dereg id/root` while logged in as root

deregistering works even for currently logged in user. this shouldn't be the default behavior.

<hr>

**Reported by:** @ongspxm
**Severity:** `High`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#348]</sub>

### LINKED ISSUE #227: [PE Dry Run]Command format for logout is invalid
**Instructions:** Logs out of the account.
Format: lo

Expected: You have been logged out

Actual: Unknown command
screenshot 2018-11-02 at 4 29 43 pm

The figure above shows that lo is an invalid format for logging out, which does not match the User Guide

<hr>

**Reported by:** @afiqlattif
**Severity:** `Low`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#417]</sub>

### LINKED ISSUE #236: [PE Dry Run]Success message shown on invalid command
[F10-4]
screen shot 2018-11-02 at 4 45 18 pm

invalid command `list`(not on user guide) gives success message. 

<hr>

**Reported by:** @jiaxiu11
**Severity:** `Medium`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#733]</sub>

### LINKED ISSUE #251: Clear command history when a user logs out
From security point of view, the command history should be cleared so that the next user will not be able to press the up and down arrow keys to view the login password of previous users.

### PR DESCRIPTION
**Pull request by AZhiKai**
Resolves #129 
Resolves #251 
Fixes #227, and fixes #219, and fixes #225, and fixes #236

Done with the refactoring stuffs, left with fixing up bugs reported in dry-run PE and removing all mentions of "persons" in comments and fxml.

_Removing code and refactoring test cases isn't for the faint-hearted. 😓_ 

System tests to be improved by @yican95 , specifically for `DeleteItemByName` and deleting a range of indexes. Currently, the system test `DeleteCommandSystemTest` only test deleting 1 item. So you will have to modify the method(s) in the class to cater for testing those other stuffs.

### CODE REVIEW

**Comment by AZhiKai** [comment]
So far all the issues raised in the first post have been handled.

File: src/test/java/seedu/restaurant/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -308,18 +289,16 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
     }
 
     /**
-     * Executes the given {@code command}, confirms that <br>
-     * - a {@code CommandException} is thrown <br>
-     * - the CommandException message matches {@code expectedMessage} <br>
-     * - the restaurant book and the filtered person list in the {@code actualModel} remain unchanged <br>
-     * - {@code actualCommandHistory} remains unchanged.
+     * Executes the given {@code command}, confirms that <br> - a {@code CommandException} is thrown <br> - the
+     * CommandException message matches {@code expectedMessage} <br> - the restaurant book and the filtered person list
```
Comment (HyperionNKJ) [inline src/test/java/seedu/restaurant/logic/commands/CommandTestUtil.java]: filtered item list

File: src/test/java/seedu/restaurant/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -329,37 +308,37 @@ public static void assertCommandFailure(Command command, Model actualModel, Comm
...
-            assertEquals(expectedFilteredList, actualModel.getFilteredPersonList());
+            assertEquals(expectedFilteredList, actualModel.getFilteredItemList());
             assertEquals(expectedCommandHistory, actualCommandHistory);
         }
     }
 
     /**
-     * Updates {@code model}'s filtered list to show only the person at the given {@code targetIndex} in the
-     * {@code model}'s restaurant book.
+     * Updates {@code model}'s filtered list to show only the person at the given {@code targetIndex} in the {@code
```
Comment (HyperionNKJ) [inline src/test/java/seedu/restaurant/logic/commands/CommandTestUtil.java]: to show only the item at the

File: src/main/java/seedu/restaurant/logic/commands/account/LogoutCommand.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Logout of the system. "
             + "Example: " + COMMAND_WORD;
 
-    public static final String MESSAGE_SUCCESS = "You have been logged out";
+    public static final String MESSAGE_SUCCESS = "You have been logged out and history is cleared";
```
[the remark this answers is not in the document]
Comment (AZhiKai) [reply src/main/java/seedu/restaurant/logic/commands/account/LogoutCommand.java]: Reverted back to original for now

[the remark this answers is not in the document]
Comment (AZhiKai) [reply src/main/java/seedu/restaurant/logic/commands/account/LogoutCommand.java]: Yup that sounds like a better idea. Will do it in another PR. Too much for this PR already. :]

**Pull request merged**